os/drivers/lcd/st7701.c

```diff
@@ -0,0 +1,368 @@
+/****************************************************************************
+ *
+ * Copyright 2024 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+#include <tinyara/lcd/lcd.h>
+#include <tinyara/lcd/lcd_dev.h>
+#include <tinyara/lcd/st7701.h>
+#include <tinyara/mipidsi/mipi_dsi.h>
+#include <tinyara/mipidsi/mipi_display.h>
+#include <debug.h>
+#include <assert.h>
+
+#define REGFLAG_DELAY                       0xFC
+#define REGFLAG_END_OF_TABLE                0xFD	// END OF REGISTERS MARKER
+
+struct st7701_dev_s {
+	/* Publicly visible device structure */
+
+	struct lcd_dev_s dev;
+
+	FAR struct mipi_dsi_device *dsi_dev;
+
+	u8 *lcd_img_buffer[MAX_NO_PLANES];
+	u8 *BackupLcdImgBuffer;
+	int fb_alloc_count;
+
+	struct st7701_config_s *config;
+};
+
+static struct st7701_dev_s g_lcdcdev;
+
+typedef struct lcm_setting_table {
+	u8 cmd;
+	u8 count;
+	u8 para_list[128];
+} lcm_setting_table_t;
+
+static lcm_setting_table_t ST7701S_init_cmd_g[] = {	/* DCS Write Long */
```

Comment:
As I have checked, this is consuming up much RAM, around ~5kB, since we are using it only for initialization phase, it should be changed to dynamic allocation, so we can free up the space after initializing the LCD. Please check if a change is necessary.

Reply:
Or maybe it can be changed to constant, so it only consumes Flash? Since the values are used to initial LCD without modifying its original values.